### PR TITLE
Allow log level specification in YML config, resolves #110

### LINKED ISSUE #110: Add log level to config
Log level for Log4j

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

**Comment by simonkorz** [comment]
Auto format the whole config file, really?

**Comment by kelunik** [comment]
Autoformat didn't work, manually formatted to two spaces. I can revert that to 4 spaces if necessary.

**Comment by marcelhollerbach** [comment]
Or maybe 3 or 5 ? srly, just leave stuff how it is.

**Comment by kelunik** [comment]
Squashed commits into one.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/config/Config.java [hunk 1]
```diff
@@ -17,17 +20,24 @@
     public Deployment deployment;
 
     /**
-     * all specified platforms in the config
+     * Platform configuration.
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/config/Config.java]: configuration>s< its a array.

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/config/Config.java]: and please no "."

File: src/main/resources/config.sample.yml [hunk 2]
```diff
@@ -45,30 +48,45 @@ platforms :
...
+
       # the id of the pybossa project to use for all experiments
       projectId: 1
+
       # if calibrations are allowed on this platform
       calibrationsAllowed: true
 
 # basic configurations for the mail-module
-mail :
+mail:
```
Comment (marcelhollerbach) [inline src/main/resources/config.sample.yml]: really, its not about this line, its about all changes to the file, do you really need to force everybodys on your personal prefered style? Can you please stop reformatting like that.

Comment (TheRittler) [inline src/main/resources/config.sample.yml]: This is a config file, that gets seen by the administrator of crowdControl. I think its important, that at least the config file hast unified formatting and style. The impression of the software would be much better, if style is unified.

Comment (marcelhollerbach) [inline src/main/resources/config.sample.yml]: 1. do you really want to tell me that a user is impressed if the style is unified ? I guess the administrator will be proud if you managed to configure the software and understand the difference between workerLocalUrl and workerPublicUrl or the fact that notificationsEmail is not used by notifications :>
2. I said its _NOT_ about this line.
3. Yes it should be consistent, is it now: NO, is it after this patch: NO.

Comment (TheRittler) [inline src/main/resources/config.sample.yml]: no, a user is not impressed if the style is unified, but its a bad impression if it looks like the right hand doesn't know what the left hand does.

Comment (marcelhollerbach) [inline src/main/resources/config.sample.yml]:
```
#hello
a:b
#next op
a:b
```

```
#hello
a:b

#next op
a:b
```

I dont really care if there is a newline or not. And anyway, if fixing it fix it consistent.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/config/Config.java (hunk 1 above)
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/config/Config.java]: Done.

File: src/main/resources/config.sample.yml (hunk 2 above)
Comment (kelunik) [inline src/main/resources/config.sample.yml]: I just made it consistent and easy to read. Whitespace is really a good thing for configuration files.

Comment (marcelhollerbach) [inline src/main/resources/config.sample.yml]: its still not consistent. Whitespaces are a personal style, so we are back at the beginning statement.

Comment (kelunik) [inline src/main/resources/config.sample.yml]: We had whitespace in front of all other documented options already...

Comment (marcelhollerbach) [inline src/main/resources/config.sample.yml]: Sorry I meant newlines. 
And no you didnt do it infront of _ALL_ other documented options, see line 36 in the file there is still " platforms :" thats why I say its still _NOT_ consistent.

Comment (kelunik) [inline src/main/resources/config.sample.yml]: Better now?

**Comment by marcelhollerbach** [comment]
yeah, go for it!

**Pull request merged**